context_manager_and_exceptions/hw_cont_manager_exc.py

```diff
@@ -0,0 +1,88 @@
+import os
+import time
+from contextlib import contextmanager, ContextDecorator
+from itertools import permutations
+
+
+# Задача-1
+# Создать объект менеджера контекста который будет переходить в
+# папку которую он принимает на вход. Так же ваш объект должен принимать
+# исключение которое он будет подавлять Если флаг об исключении отсутствует, исключение должно быть поднято.
+
+
+class MyContextManager():
+    """Сlass implements the context manager structure for changing the current directory to the working directory"""
+
+    def __init__(self, dirname, type_exception):
+        self._work_dir_path = os.path.abspath(dirname)
+        self._type_exception = type_exception
+        self._current_dir_path = os.getcwd()
+
+    def __enter__(self):
+        if not os.path.isdir(self._work_dir_path):
+            os.mkdir(self._work_dir_path)
+        os.chdir(self._work_dir_path)
+        print("Changed directory")
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        os.chdir(self._current_dir_path)
+        if exc_type is not None and issubclass(exc_type, self._type_exception):
+            print("Directory not found")
+            return True
```

Comment:
this task was implemented incorrectly, you should send an error to the context manager.
Example :

with cd('tmp', FileNotFoundError) that FileNotFoundError should be suppressed
or
with cd('tmp.txt', NotADirectoryError) that NotADirectoryError should be suppressed

Creating a directory if path doesn't exists is ok but not for this task